frontend/catalyst/compiler.py

```diff
@@ -453,11 +453,14 @@ def run_from_ir(self, ir: str, module_name: str, workspace: Directory):
         if self.options.verbose:
             print(f"[LIB] Running compiler driver in {workspace}", file=self.options.logfile)
 
+        data = ir if type(ir) != str else ir.encode("utf-8")
+        suffix = ".bc" if type(ir) != str else ".mlir"
+
         with tempfile.NamedTemporaryFile(
-            mode="w", suffix=".mlir", dir=str(workspace), delete=False
+            mode="w+b", suffix=suffix, dir=str(workspace), delete=False
```

Comment:
Does this work for both text and bytes?

Reply:
We are writing just bytes now. See above:

```python
        data = ir if type(ir) != str else ir.encode("utf-8")
```

All tests are passing.

Reply:
Good point, for some reason I read that as accepting both strings and bytes. All good 👍